hack/run-openshift

```diff
@@ -0,0 +1,183 @@
+    if [[ "$(oc get pods --no-headers=true -l${label} | grep "1/1" | wc -l)" -eq ${count} ]]; then
+      return 0
+    fi
+    echo "Waiting for ${count} ready pods labeled with '${label}'..."
+    sleep 20
+  done
+  return 1
+}
+
+# create_volumes reates Kubernetes volumes with given name prefix, specified capacity and accessMode
+function create_volumes() {
+  local number=${1}
+  local name=${2}
+  local capacity=${3}
+  local accessMode=${4}
+
+  for i in $(seq ${number}); do
+    local path=$(mktemp -d)
+    chmod a+rwx ${path}
+    chcon -R -t svirt_sandbox_file_t ${path}
+
+    echo "kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: ${name}${i}
+  labels:
+    type: local
+spec:
+  capacity:
+    storage: ${capacity}
+  accessModes:
+    - ${accessMode}
+  hostPath:
+    path: '${path}'" | docker exec -i origin kubectl create -f -
+  done
+}
+
+#
+# Project specific functions
+#
+
+# mongo_admin_cmd evals JS commands as admin on CONTAINER_IP host
+function mongo_admin_cmd() {
+  mongo admin --host $CONTAINER_IP -u admin -p "$ADMIN_PASS" --eval "${@}"
+}
+
+# insert_and_wait_for_replication insert data in host and wait all replset members
+# applied recent oplog entry
+function insert_and_wait_for_replication() {
+  local host
+  host=$1
+  local data
+  data=$2
+
+  # Storing document into replset and wait replication to finish
+  CONTAINER_IP="${host}" \
+  mongo_admin_cmd "db.getSiblingDB('test_db').data.insert(${data});
+    var i = 60;
+    while(i > 0) {
+      var status=rs.status();
+      var optime=status.members[0].optime;
+      var ok=true;
+      for(var i=1; i < status.members.length; i++) {
+        if(tojson(optime) != tojson(status.members[i].optime)) {
+          ok=false
+        }
+      };
+      if(ok == true) {
+        print('INFO: All members of replicaset are synchronized')
+        break;
+      }
+      sleep(1);
+      i--;
+    }"
+}
+
+# wait_replicaset_members waits till replset has specified number of members
+function wait_replicaset_members() {
+  local host
+  host=$1
+  local count
+  count=$2
+
+  local script
+  script="for (var i = 0; i < 60; i++) {
+    var ret = rs.status().members.length;
+    if (ret == ${count}) {
+      quit(0);
+    }
+    sleep(1);
+  }
+  print('ERROR: Wrong count of members in replicaset...');
+  printjson(rs.status());
+  quit(1);"
+
+  if ! CONTAINER_IP="${host}" mongo_admin_cmd "${script}"; then
+    return 1
+  fi
+  return 0
+}
+
+# Deploy MongoDB clustered application
+USER=user
+PASS=pass
+ADMIN_PASS=adminPass
+DB=db
+
+create_volumes 3 "mongo-data" "500M" "ReadWriteOnce"
+
+oc new-project petset-example
+oc new-app --file=https://raw.githubusercontent.com/sclorg/mongodb-container/cf99d38bfcff2657fd1effba7c2bf69e6bc96dfd/examples/petset/mongodb-petset-persistent.yaml -p MONGODB_USER=${USER} -p MONGODB_PASSWORD=${PASS} -p MONGODB_DATABASE=${DB} -p MONGODB_ADMIN_PASSWORD=${ADMIN_PASS} -p MONGODB_IMAGE=${IMAGE_NAME} -p VOLUME_CAPACITY=500M
```

Comment:
i'd reference the latest file (especially once you move to the new oc cluster up which should have the new statefulset api) to ensure the template doesn't get broken in the future.